### ISSUE TITLE
Improve terraform-provider-tarmak PoC by a net/rpc approach

### ISSUE THREAD

**Issue by simonswine** [jetstack/tarmak#18]
While we are working on having a proper design on a tarmak terraform provider agreed on. I would like to get a more detailed PoC than the one I started:

https://gitlab.jetstack.net/christian.simon/terraform-provider-tarmak

As part of this issue you should implement a terraform provider with a single resource. An example tf could look like this:

```
resource `tarmak_init_token` `worker` {
  environment = 'josh'
  cluster = 'cluster'
  role = 'worker'
}

resource `null_resource` `worker_token_to_file` {
  provisioner "local-exec" {
    command = "echo ${tarma_init_token.worker.init_token} >> token_test.txt"
}
```

For that simple example the token should go through this flow:

```
<<tarmak_fake| listen to a unix socket with `net/rpc`>>

<-- `net/rpc` --| 

<<terraform-provider-tarmak | blocks till it's able to connect to the unix socket>>
```

So basically tarmak_fake provides an net/rpc server exposing a method `tarmakInitToken(env string, cluster string, role string) (initToken string, err error)`.

The terraform-provider-tarmak is blocking till it's able to speak to tarmak_fake and then call through 
`net/rpc` `tarmakInitToken` and reports back the output `init_token`

<hr/>

For a later iteration we need to use stdin/stdout for RPC communications once we run terraform + ut terraform-provider-tarmak in docker. This is something that someone else trailed here:to 
https://gist.github.com/nicolai86/0dee36c4250ca1a9430bb7ebec4cfa58
(just to have that link somewhere :slightly_smiling_face:)

**Comment by JoshVanL** [comment]
Changed id to hash of init_token and changed values to variables in main.tf
Started to implement using Stdio/Stdo for rpc communication (WIP). Pushed to save progress.

**Comment by simonswine** [comment]
The implementation is continued in #110
